fetchcode/purl2url/__init__.py

```diff
@@ -0,0 +1,127 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from urllib.parse import urlparse
+
+
+def index_in_list(index, list):
+    return index < len(list)
+
+
+def build_bitbucket(name, namespace, version, qualifiers, subpath):
+
+    if name and namespace:
+        url = "https://bitbucket.org/{namespace}/{name}".format(
+            namespace=namespace, name=name
+        )
+        if version:
+            url = "{url}/src/{version}".format(url=url, version=version)
+        if subpath:
+            url = "{url}/{subpath}".format(url=url, subpath=subpath)
+        return url
+
+
+def build_cargo(name, namespace, version, qualifiers, subpath):
+    if name and version:
+        return "https://crates.io/api/v1/crates/{name}/{version}/download".format(
+            name=name, version=version
+        )
+
+
+def build_github(name, namespace, version, qualifiers, subpath):
+
+    if name and namespace:
```

Comment:
Why not check the inverse of this and return right away? This would remove this big indented block.